### DISCUSSION TITLE
Access iTexture properties

### DISCUSSION THREAD

**Discussion by Samuel780000** [VentuzTechnology/Ventuz.Extension.prerelease#32] [Q&A]
Hi everyone !

How to access input iTextures properties such as Width, Height, GetPixels, etc... ?

**Comment by VentuzRalfStanke** [comment]
Hi Samual,
currently there is no API for ITexture yet. The textures in our engine are of different types (of course). Not all of them can be treated as an image, so the properties you mentioned above are not always applicable for any texture. (e.g. 3D textures, interlaced video, direct DMA transferred, etc). But anyway, 95% of the textures are actually images - so we can put an API to our wish-list.
If you plan to perform "read-backs" to cpu system memory (GetPixel) you have to have in mind, that this is usually a very slow procedure and may the renderer to stutter and interfere with PCIe performance optimizations. A proper API for it is not easy to implement. But again - we put it on our wish-list.

Hope that helped,

Ralf

**Comment by Samuel780000** [comment]
When i follow the example to generate Texture, I see that there is a way Bitmap => Texture, but not Texture=> Bitmap.

https://github.com/VentuzTechnology/Ventuz.Extension.prerelease/wiki/Documentation#texture
